subprojects/language-native/src/integTest/groovy/org/gradle/language/swift/SwiftCachingIntegrationTest.groovy

```diff
@@ -0,0 +1,138 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.language.swift
+
+import org.gradle.integtests.fixtures.DirectoryBuildCacheFixture
+import org.gradle.nativeplatform.fixtures.AbstractInstalledToolChainIntegrationSpec
+import org.gradle.nativeplatform.fixtures.app.SwiftAppWithLibraries
+import org.gradle.test.fixtures.file.TestFile
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
+import spock.lang.Unroll
+
+@Requires(TestPrecondition.SWIFT_SUPPORT)
+class SwiftCachingIntegrationTest extends AbstractInstalledToolChainIntegrationSpec implements DirectoryBuildCacheFixture {
+
+    def app = new SwiftAppWithLibraries()
+
+    def setupProject(TestFile project = temporaryFolder.testDirectory) {
+        project.file('settings.gradle') << '''
+            include 'hello', 'log'
+            rootProject.name = 'app'
+        '''
+        project.file('settings.gradle') << localCacheConfiguration()
+        project.file('build.gradle').text = '''
+            apply plugin: 'swift-executable'
+            dependencies {
+                implementation project(':hello')
+            }
+
+            project(':hello') {
+                apply plugin: 'swift-library'
+                dependencies {
+                    api project(':log')
+                }
+            }
+            project(':log') {
+                apply plugin: 'swift-library'
+            }
+        '''
+        app.logLibrary.writeToProject(project.file('log'))
+        app.library.writeToProject(project.file('hello'))
+        app.executable.writeToProject(project)
+    }
+
+    def 'compilation can be cached'() {
+        setupProject()
+
+        when:
+        withBuildCache().run compileTask(buildType)
+
+        then:
+        compileIsNotCached(buildType)
+
+        when:
+        withBuildCache().run 'clean', installTask(buildType)
+
+        then:
+        compileIsCached(buildType)
+        installation("build/install/main/${buildType.toLowerCase()}").exec().out == app.expectedOutput
+
+        where:
+        buildType << ['Debug', 'Release']
+    }
+
+    @Unroll
+    def "compilation task is relocatable (#buildType)"() {
+        def originalLocation = file('original-location')
+        def newLocation = file('new-location')
+        def newInstallDir = newLocation.file("build/install/main/${buildType.toLowerCase()}")
+        setupProject(originalLocation)
+        setupProject(newLocation)
+
+        when:
+        inDirectory(originalLocation)
+        withBuildCache().run compileTask(buildType), installTask(buildType)
+
+        then:
+        compileIsNotCached(buildType)
+
+        when:
+        originalLocation.deleteDir()
+        executer.beforeExecute {
+            inDirectory(newLocation)
+        }
+        withBuildCache().succeeds ":hello${compileTask(buildType)}", ":log${compileTask(buildType)}"
```

Comment:
We should check if the dependency tasks are actually loaded from cache here.

Reply:
It would be nice to refactor this line to:

```groovy
withBuildCache().succeeds compileTask(":hello", buildType), compileTask(":log", buildType)
```

Reply:
I checked the dependent tasks and refactored to what you suggested.